cartridges/int_algolia_sfra/cartridge/static/default/js/algolia/instantsearch-config.js

```diff
@@ -368,6 +380,17 @@ function enableInstantSearch(config) {
                                     indexName: item.__indexName,
                                 });
                             });
+                            if (colorFacets.length > 0) {
+                                const selectedColorVariation = item.colorVariations.find(i => {
+                                    return colorFacets.includes(i.color);
+                                });
+                                if (selectedColorVariation) {
+                                    item.image = {
```

Comment:
The image is already set above. There is no need to look into facets or anything complicated for the variant-level model: **the image in the record is the one we want to display**